tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/dataprovider/DataProviderManager.java

```diff
@@ -231,6 +232,38 @@ public List<IDataProviderDescriptor> getAvailableProviders(@Nullable ITmfTrace t
         return list;
     }
 
+    /**
+     * Get the list of available providers for this trace / experiment without
+     * triggering the analysis or creating the provider
+     *
+     * @param trace
+     *            queried trace
+     * @param configId
+     *            a configuration ID used for the data provider
+     * @param ids
+     *            A list of ID of the data provider factory or a data provider ID of
+     *            form factoryId:secondaryId
+     * @return list of the available providers for this trace / experiment
+     * @since 9.5
+     */
+    public synchronized @NonNull List<IDataProviderDescriptor> getAvailableProviders(ITmfTrace trace, String configId, String... ids) {
```

Comment:
For supporting per base DataProvider conifgurations We would need a `getAvailableProviders(ITmfTrace trace, String... ids)` to return all DataProviders which contain the (factoryId == baseDataProvider ID).
That, or DataProvider Descriptor object would need to be a tree of descriptor strings.

Reply:
That could be useful. It would return all derived data providers for all configurations.